libnmstate/nispor/ethernet.py

```diff
@@ -26,3 +27,28 @@ class NisporPluginEthernetIface(NisporPluginBaseIface):
     @property
     def type(self):
         return InterfaceType.ETHERNET
+
+    def to_dict(self):
+        info = super().to_dict()
+        if self.np_iface.sr_iov:
+            vf_infos = []
+            for vf in self.np_iface.sr_iov.vfs:
+                vf_info = {
+                    Ethernet.SRIOV.VFS.ID: vf.vf_id,
+                    Ethernet.SRIOV.VFS.MAC_ADDRESS: vf.mac,
+                    Ethernet.SRIOV.VFS.SPOOF_CHECK: vf.spoof_check,
+                    Ethernet.SRIOV.VFS.TRUST: vf.trust,
+                    Ethernet.SRIOV.VFS.MIN_TX_RATE: vf.min_tx_rate,
+                    Ethernet.SRIOV.VFS.MAX_TX_RATE: vf.max_tx_rate,
+                }
+                vf_infos.append(vf_info)
+
+            sriov_info = {
+                Ethernet.SRIOV.TOTAL_VFS: len(self.np_iface.sr_iov.vfs),
+                Ethernet.SRIOV.VFS_SUBTREE: vf_infos,
+            }
+            info[Ethernet.CONFIG_SUBTREE] = {
```

Comment:
This is incorrect according to `schemas/operational-state.yaml`.
The schema file put `sr-iov` at the top level of `iface_info`.
While the `nm/sriov.py` use `Ethernet.CONFIG_SUBTREE`.

Please create a separate PR fixing this first.

Reply:
Done on #1323